### PR TITLE
M2P-640 Add support for Mageworx rewards

### PR DESCRIPTION
**Pull request by dumega**
# Description
Add support for Mageworx rewards

Fixes: https://boltpay.atlassian.net/browse/M2P-640

#changelog Add support for Mageworx rewards

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: ThirdPartyModules/MageWorx/RewardPoints.php [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            ];
+            if ($mageWorxRewardPointsHelper->isAllowedCustomPointsAmount()) {
+                $jsLayout["mageworx-reward-points"]["config"] = [
+                    'template' => 'Bolt_Boltpay/third-party-modules/mageworx/reward-points/cart/rewardpoints_custom_amount'
+                ];
+            } else {
+                $jsLayout["mageworx-reward-points"]["config"] = [
+                    'template' => 'Bolt_Boltpay/third-party-modules/mageworx/reward-points/cart/rewardpoints'
+                ];
+            }
```
Comment (danac-gs) [inline ThirdPartyModules/MageWorx/RewardPoints.php:181]:
```suggestion
                ],
                "config" => [
                    "template" => $mageWorxRewardPointsHelper->isAllowedCustomPointsAmount()
                        ? 'Bolt_Boltpay/third-party-modules/mageworx/reward-points/cart/rewardpoints_custom_amount'
                        : 'Bolt_Boltpay/third-party-modules/mageworx/reward-points/cart/rewardpoints'
                ]
            ];
```

**Review by danac-gs — APPROVED** [review]
LGTM, one suggestion

File: ThirdPartyModules/MageWorx/RewardPoints.php (hunk 1 above)
Comment (dumega) [reply ThirdPartyModules/MageWorx/RewardPoints.php]: Done, please review again

File: ThirdPartyModules/MageWorx/RewardPoints.php [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+                    "errors" => [
+                        "sortOrder"   => 0,
+                        "component"   => "MageWorx_RewardPoints/js/view/payment/rewardpoints-messages",
+                        "displayArea" => "messages",
+                    ]
+                ]
+            ];
+            $jsLayout["mageworx-reward-points"]["config"] = [
+                "template" => $mageWorxRewardPointsHelper->isAllowedCustomPointsAmount()
+                            ? 'Bolt_Boltpay/third-party-modules/mageworx/reward-points/cart/rewardpoints_custom_amount',
```
Comment (danac-gs) [inline ThirdPartyModules/MageWorx/RewardPoints.php:175]:
```suggestion
                            ? 'Bolt_Boltpay/third-party-modules/mageworx/reward-points/cart/rewardpoints_custom_amount'
```
this comma throws a syntax error, see the failing tests

**Review by danac-gs — CHANGES_REQUESTED** [review]

File: ThirdPartyModules/MageWorx/RewardPoints.php (hunk 2 above)
Comment (dumega) [reply ThirdPartyModules/MageWorx/RewardPoints.php]: OH :( 

Fixed. Thankls

**Review by danac-gs — APPROVED** [review]

**Pull request merged**